Mathlib/RingTheory/JacobsonIdeal.lean

```diff
@@ -427,6 +427,30 @@ theorem isPrimary_of_isMaximal_radical [CommRing R] {I : Ideal R} (hi : IsMaxima
         exact I.sub_mem (I.mul_mem_left _ hxy) (I.mul_mem_left _ hz))
       (this ▸ id)⟩
 
+lemma isUnit_or_mem_jacobson_bot_of_dvd_isNilpotent_of_zero_divisors_mem_jacobson_bot [CommRing R]
+    {s a : R} (hsn : s ≠ 0) (hs : IsNilpotent s) (hd : a ∣ s)
+    (hj : (⊥ : Ideal R).jacobson ≤ Ideal.span {s})
+    (h0 : {x : R | ∃ y ≠ 0, y * x = 0} ⊆ (⊥ : Ideal R).jacobson) :
+    IsUnit a ∨ a ∈ (⊥ : Ideal R).jacobson := by
```

Comment:
The statement of this lemma looks very awkward to me. It seems to me that `hs` implies that `s` is a zero-divisor, so `hj` and `h0` imply Jacobson radical = zero-divisors = span {s}. Since minimal primes are contained in zero-divisors, this implies that `s` is prime, which then implies `s` is irreducible which is your goal.

I would suggest having a lemma `{x : R | ∃ y ≠ 0, y * x = 0} <= nilradical R => IsPrime (nilRadical R)` instead.